### PR TITLE
adds account ID resource

### PR DESCRIPTION
**Pull request by xh3b4sd**
This PR suggests a new `accountid` resource. It is pretty simple. It only fetches the AWS Account ID once and puts it into the controller context. This saves about 6 calls to AWS STS during one reconciliation loop.

### CODE REVIEW

File: service/controller/v22/encrypter/vault/vault.go [hunk 1]
```diff
@@ -152,21 +152,16 @@ func (e *Encrypter) EnsureCreatedAuthorizedIAMRoles(ctx context.Context, customO
 		return microerror.Mask(err)
 	}
 
-	ctlCtx, err := controllercontext.FromContext(ctx)
+	cc, err := controllercontext.FromContext(ctx)
```
Comment (kopiczko) [inline service/controller/v22/encrypter/vault/vault.go]: We just named it `ctlCtx` in @tomahawk28 PR. Would be good to get some consensus in fmt.

Comment (kopiczko) [reply service/controller/v22/encrypter/vault/vault.go]: https://github.com/giantswarm/fmt/pull/42

Comment (rossf7) [reply service/controller/v22/encrypter/vault/vault.go]: 42 <3  I voted for `ctlCtx`.

Comment (xh3b4sd) [reply service/controller/v22/encrypter/vault/vault.go]: We can change once we know how.

**Pull request merged**